messaging/src/main/java/org/axonframework/commandhandling/distributed/DistributedCommandBus.java

```diff
@@ -177,25 +160,27 @@ public <C, R> void dispatch(@Nonnull CommandMessage<C> command,
         CommandMessage<? extends C> interceptedCommand = intercept(command);
         MessageMonitor.MonitorCallback messageMonitorCallback = messageMonitor.onMessageIngested(interceptedCommand);
         Optional<Member> optionalDestination = commandRouter.findDestination(interceptedCommand);
-        if (optionalDestination.isPresent()) {
-            Member destination = optionalDestination.get();
-            try {
+        Span span = spanFactory.createDispatchSpan(() -> "DistributedCommandBus.dispatch", command).start();
+        try (SpanScope ignored = span.makeCurrent()) {
+            if (optionalDestination.isPresent()) {
+                Member destination = optionalDestination.get();
                 connector.send(destination,
                                interceptedCommand,
```

Comment:
I think the `CommandMessage` given to the `CommandBusConnector` should have the span context propagated too.
So, I would expect a `SpanFactory#propagateContext` invocation, where the result is given to the connector.

Reply:
added propagation